components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -2729,4 +2729,153 @@ void RewardsServiceImpl::GetExcludedPublishersNumberDB(
+      base::Bind(&RewardsServiceImpl::GetCountByReconcileStamp,
+        AsWeakPtr(), callback));
+}
+
+uint64_t AutoContributionLastReconcileDateOnFileTaskRunner(
+    PublisherInfoDatabase* backend) {
+  return backend->GetLastReconcileDate();
+}
+
+void RewardsServiceImpl::GetLastReconcileStamp(
+    const rewards_counter::OnRewardsCountedCallback callback,
+    int64_t count) {
+  base::PostTaskAndReplyWithResult(
+      file_task_runner_.get(),
+      FROM_HERE,
+      base::Bind(&AutoContributionLastReconcileDateOnFileTaskRunner,
+          publisher_info_backend_.get()),
+      base::Bind(&RewardsServiceImpl::OnRewardsCounted,
+          AsWeakPtr(),
+          callback,
+          count));
+}
+
+void RewardsServiceImpl::GetCountByReconcileStamp(
+    const rewards_counter::OnRewardsCountedCallback callback,
+    uint64_t reconcile_stamp) {
+  base::PostTaskAndReplyWithResult(
+      file_task_runner_.get(),
+      FROM_HERE,
+      base::Bind(&AutoContributionCountOnFileTaskRunner,
+          publisher_info_backend_.get(), reconcile_stamp),
+      base::Bind(&RewardsServiceImpl::GetLastReconcileStamp,
+          AsWeakPtr(),
+          callback));
+}
+
+void RewardsServiceImpl::OnRewardsCounted(
+    rewards_counter::OnRewardsCountedCallback callback,
+    int64_t count,
+    uint64_t previous_reconcile_stamp) {
+  base::string16 previous_reconcile_date;
+  if (previous_reconcile_stamp != 0) {
+    previous_reconcile_date = TimeFormatShortDate(
+        base::Time::FromDoubleT(previous_reconcile_stamp));
+  }
+  callback.Run(count, previous_reconcile_date);
+}
+
+void RewardsServiceImpl::RemoveAutoContributeData(
+    OnRemoveRewardsDataCallback callback) {
+  GetReconcileStamp(
+      base::Bind(&RewardsServiceImpl::ClearAutoContributeData,
+        AsWeakPtr(), callback));
+}
+
+bool RemoveAutoContributeEntriesOnFileTaskRunner(
+    PublisherInfoDatabase* backend, uint64_t reconcile_stamp) {
+  return backend->RemoveAutoContributeEntries(reconcile_stamp);
+}
+
+void RewardsServiceImpl::ClearAutoContributeData(
+    OnRemoveRewardsDataCallback callback,
+    uint64_t reconcile_stamp) {
+  base::PostTaskAndReplyWithResult(
+    file_task_runner_.get(),
+    FROM_HERE,
+    base::Bind(&RemoveAutoContributeEntriesOnFileTaskRunner,
+        publisher_info_backend_.get(), reconcile_stamp),
+    base::Bind(&RewardsServiceImpl::OnClearAutoContributeData,
+        AsWeakPtr(), callback));
+}
+
+void RewardsServiceImpl::OnClearAutoContributeData(
+    OnRemoveRewardsDataCallback callback,
+    bool result) {
+  for (auto& observer : observers_) {
+    observer.OnAutoContributeDataRemoved(this);
+  }
+  callback.Run(result);
+}
+
+bool RemoveAllRewardsEntriesOnFileTaskRunner(PublisherInfoDatabase* backend) {
+  return backend->RemoveAllRewardsEntries();
+}
+
+void RewardsServiceImpl::RemoveRewardsData(
+    int remove_mask,
+    OnRemoveRewardsDataCallback callback) {
+  if (remove_mask & ClearRewardsDataType::REWARDS_TYPE_AUTO_CONTRIBUTE) {
```

Comment:
these operations should be called together synchronously on the task runner thread with a single callback

Reply:
Done.